research-domain-agent/src/test/java/care/smith/fts/rda/impl/DeidentifhirStepIT.java

```diff
@@ -1,75 +1,110 @@
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
 import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
 import static com.typesafe.config.ConfigFactory.parseResources;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.http.HttpHeaders.CONTENT_TYPE;
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON_VALUE;
 import static reactor.test.StepVerifier.create;
 
 import care.smith.fts.api.TransportBundle;
 import care.smith.fts.rda.services.deidentifhir.DeidentifhirUtil;
+import care.smith.fts.test.connection_scenario.AbstractConnectionScenarioIT;
 import care.smith.fts.util.WebClientFactory;
+import com.github.tomakehurst.wiremock.client.MappingBuilder;
 import com.github.tomakehurst.wiremock.client.WireMock;
 import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
 import com.github.tomakehurst.wiremock.junit5.WireMockTest;
 import io.micrometer.core.instrument.MeterRegistry;
 import java.io.IOException;
+import java.util.stream.Stream;
 import org.hl7.fhir.r4.model.Bundle;
 import org.hl7.fhir.r4.model.Resource;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
+import reactor.core.publisher.Mono;
 
 @SpringBootTest
 @WireMockTest
-class DeidentifhirStepTest {
+class DeidentifhirStepIT extends AbstractConnectionScenarioIT {
+
   @Autowired MeterRegistry meterRegistry;
-  private DeidentifhirStep step;
   private WireMock wireMock;
 
+  private DeidentifhirStep step;
+  private Bundle bundle;
+
+  @Override
+  protected Stream<TestStep<?>> createTestSteps() {
+    return Stream.of(
+        new TestStep<Bundle>() {
+          @Override
+          public MappingBuilder getBuilder() {
+            return DeidentifhirStepIT.getBuilder();
+          }
+
+          @Override
+          public Mono<Bundle> executeStep() {
+            return step.deidentify(new TransportBundle(bundle, "transferId"));
+          }
+
+          @Override
+          public String acceptedContentType() {
+            return APPLICATION_FHIR_JSON;
+          }
+        });
+  }
+
   @BeforeEach
-  void setUp(WireMockRuntimeInfo wireMockRuntime, @Autowired WebClientFactory clientFactory) {
+  void setUp(WireMockRuntimeInfo wireMockRuntime, @Autowired WebClientFactory clientFactory)
+      throws IOException {
     var config = parseResources(DeidentifhirUtil.class, "TransportToRD.profile");
     var client = clientFactory.create(clientConfig(wireMockRuntime));
     step = new DeidentifhirStep(config, client, meterRegistry);
     wireMock = wireMockRuntime.getWireMock();
+    bundle = generateOnePatient("tid1", "2024", "identifierSystem");
   }
 
   @AfterEach
   void tearDown() {
     wireMock.resetMappings();
   }
 
+  private static MappingBuilder getBuilder() {
```

Comment:
getBuilder() sounds very generic. Also it is used in one place only, could be inlined.